Fix for #436. Disable the continuous trait constraint due to conflicts

bruttenberg opened the pull request:
with likelihood weighting. Also removed MH tests with conditions in
continuous tests because they will never pass now that the constraint is
removed.

apfeffer commented:
Thanks. The next step is to put the check directly in MH so these queries work again.

+  +  +  +  +  +  +  +  +  +  +  +
Avi Pfeffer
Charles River Analytics Inc.
apfeffer@cra.com<mailto:apfeffer@cra.com>
(617) 491-3474x513

From: bruttenberg [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 9:44 AM
To: p2t2/figaro
Subject: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)


with likelihood weighting. Also removed MH tests with conditions in
continuous tests because they will never pass now that the constraint is
removed.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/p2t2/figaro/pull/442

Commit Summary

  *   Fix for #436. Disable the continuous trait constraint due to conflicts

File Changes

  *   M Figaro/src/main/scala/com/cra/figaro/algorithm/factored/ParticleGenerator.scala<https://github.com/p2t2/figaro/pull/442/files#diff-0> (4)
  *   M Figaro/src/main/scala/com/cra/figaro/language/Continuous.scala<https://github.com/p2t2/figaro/pull/442/files#diff-1> (6)
  *   M Figaro/src/test/scala/com/cra/figaro/test/experimental/structured/FactorMakerTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-2> (4)
  *   M Figaro/src/test/scala/com/cra/figaro/test/library/atomic/continuous/ContinuousTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-3> (15)

Patch Links:

  *   https://github.com/p2t2/figaro/pull/442.patch
  *   https://github.com/p2t2/figaro/pull/442.diff

—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442>.

bruttenberg commented:
Well, the constraint is written incorrectly at the moment (won’t work for factors as we discussed), so we don’t want to just enable it for MH. Would like a solution where the constraint is applied correctly so it can work for factors, MH, but not for importance sampling.

+  +  +  +  +  +  +  +  +  +  +  +  +  +
Brian Ruttenberg
Charles River Analytics Inc.
617.491.3474 x730
www.cra.com<http://www.cra.com/>

From: apfeffer [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 10:23 AM
To: p2t2/figaro
Cc: Brian Ruttenberg
Subject: Re: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)

Thanks. The next step is to put the check directly in MH so these queries work again.

+ + + + + + + + + + + +
Avi Pfeffer
Charles River Analytics Inc.
apfeffer@cra.com<mailto:apfeffer@cra.com<mailto:apfeffer@cra.com%3cmailto:apfeffer@cra.com>>
(617) 491-3474x513

From: bruttenberg [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 9:44 AM
To: p2t2/figaro
Subject: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)


with likelihood weighting. Also removed MH tests with conditions in
continuous tests because they will never pass now that the constraint is
removed.

________________________________
You can view, comment on, or merge this pull request online at:

https://github.com/p2t2/figaro/pull/442

Commit Summary

* Fix for #436. Disable the continuous trait constraint due to conflicts

File Changes

* M Figaro/src/main/scala/com/cra/figaro/algorithm/factored/ParticleGenerator.scala<https://github.com/p2t2/figaro/pull/442/files#diff-0> (4)
* M Figaro/src/main/scala/com/cra/figaro/language/Continuous.scala<https://github.com/p2t2/figaro/pull/442/files#diff-1> (6)
* M Figaro/src/test/scala/com/cra/figaro/test/experimental/structured/FactorMakerTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-2> (4)
* M Figaro/src/test/scala/com/cra/figaro/test/library/atomic/continuous/ContinuousTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-3> (15)

Patch Links:

* https://github.com/p2t2/figaro/pull/442.patch
* https://github.com/p2t2/figaro/pull/442.diff

—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442>.


—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442#issuecomment-95603205>.

apfeffer commented:
For factored algorithms, we would need to make sure the observed value is included in the sampled range of the element. Then, the constraint factor will simply ensure that only that value has positive probability. The non-constraint factor for the observed element will ensure that the probability of this value given the parents is correct. This will enforce the constraint correctly. So, no extra machinery is needed for sampling algorithms except to ensure that the required value is present.

So I think MH needs its own separate fix.

+  +  +  +  +  +  +  +  +  +  +  +
Avi Pfeffer
Charles River Analytics Inc.
apfeffer@cra.com<mailto:apfeffer@cra.com>
(617) 491-3474x513

From: bruttenberg [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 10:35 AM
To: p2t2/figaro
Cc: Avi Pfeffer
Subject: Re: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)

Well, the constraint is written incorrectly at the moment (won’t work for factors as we discussed), so we don’t want to just enable it for MH. Would like a solution where the constraint is applied correctly so it can work for factors, MH, but not for importance sampling.

+ + + + + + + + + + + + + +
Brian Ruttenberg
Charles River Analytics Inc.
617.491.3474 x730
www.cra.com<http://www.cra.com/<http://www.cra.com%3chttp:/www.cra.com/>>

From: apfeffer [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 10:23 AM
To: p2t2/figaro
Cc: Brian Ruttenberg
Subject: Re: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)

Thanks. The next step is to put the check directly in MH so these queries work again.

+ + + + + + + + + + + +
Avi Pfeffer
Charles River Analytics Inc.
apfeffer@cra.com<mailto:apfeffer@cra.com<mailto:apfeffer@cra.com%3cmailto:apfeffer@cra.com<mailto:apfeffer@cra.com%3cmailto:apfeffer@cra.com%3cmailto:apfeffer@cra.com%3cmailto:apfeffer@cra.com>>>
(617) 491-3474x513

From: bruttenberg [mailto:notifications@github.com]
Sent: Thursday, April 23, 2015 9:44 AM
To: p2t2/figaro
Subject: [figaro] Fix for #436. Disable the continuous trait constraint due to conflicts (#442)


with likelihood weighting. Also removed MH tests with conditions in
continuous tests because they will never pass now that the constraint is
removed.

________________________________
You can view, comment on, or merge this pull request online at:

https://github.com/p2t2/figaro/pull/442

Commit Summary

* Fix for #436. Disable the continuous trait constraint due to conflicts

File Changes

* M Figaro/src/main/scala/com/cra/figaro/algorithm/factored/ParticleGenerator.scala<https://github.com/p2t2/figaro/pull/442/files#diff-0> (4)
* M Figaro/src/main/scala/com/cra/figaro/language/Continuous.scala<https://github.com/p2t2/figaro/pull/442/files#diff-1> (6)
* M Figaro/src/test/scala/com/cra/figaro/test/experimental/structured/FactorMakerTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-2> (4)
* M Figaro/src/test/scala/com/cra/figaro/test/library/atomic/continuous/ContinuousTest.scala<https://github.com/p2t2/figaro/pull/442/files#diff-3> (15)

Patch Links:

* https://github.com/p2t2/figaro/pull/442.patch
* https://github.com/p2t2/figaro/pull/442.diff

—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442>.


—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442#issuecomment-95603205>.


—
Reply to this email directly or view it on GitHub<https://github.com/p2t2/figaro/pull/442#issuecomment-95607514>.
